bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/ChargeStateUpdater.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.saicismart.internal;
+
+import static org.openhab.binding.saicismart.internal.SAICiSMARTBindingConstants.API_ENDPOINT_V30;
+import static org.openhab.binding.saicismart.internal.SAICiSMARTBindingConstants.CHANNEL_POWER;
+import static org.openhab.binding.saicismart.internal.SAICiSMARTBindingConstants.CHANNEL_SOC;
+
+import java.net.URISyntaxException;
+import java.time.ZonedDateTime;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.bn.coders.IASN1PreparedElement;
+import org.openhab.binding.saicismart.internal.exceptions.ChargingStatusAPIException;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ThingStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.GsonBuilder;
+
+import net.heberling.ismart.asn1.v3_0.Message;
+import net.heberling.ismart.asn1.v3_0.MessageCoder;
+import net.heberling.ismart.asn1.v3_0.entity.OTA_ChrgMangDataResp;
+
+/**
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+class ChargeStateUpdater implements Callable<OTA_ChrgMangDataResp> {
+    private final Logger logger = LoggerFactory.getLogger(ChargeStateUpdater.class);
+
+    private final SAICiSMARTHandler saiCiSMARTHandler;
+
+    public ChargeStateUpdater(SAICiSMARTHandler saiCiSMARTHandler) {
+        this.saiCiSMARTHandler = saiCiSMARTHandler;
+    }
+
+    public OTA_ChrgMangDataResp call() throws URISyntaxException, ExecutionException, InterruptedException,
+            TimeoutException, ChargingStatusAPIException {
+        MessageCoder<IASN1PreparedElement> chargingStatusRequestmessageCoder = new MessageCoder<>(
+                IASN1PreparedElement.class);
+        Message<IASN1PreparedElement> chargingStatusMessage = chargingStatusRequestmessageCoder.initializeMessage(
+                saiCiSMARTHandler.getBridgeHandler().getUid(), saiCiSMARTHandler.getBridgeHandler().getToken(),
+                saiCiSMARTHandler.config.vin, "516", 768, 5, null);
```

Comment:
@lsiepel If I make this class @NonNullByDefault I can not send a null as a parameter.  If I initialize this parameter the method does not work correctly.  What is the best way to satisfy the SAT report and send a null?

Reply:
I think I found a solution with this commit: https://github.com/openhab/openhab-addons/pull/15894/commits/d934c3db9adb91338d2d7162a0dda61bc9e3db87